File: frontend/src/models/RecipeDto.ts
import type {Ingredient} from "./Ingredient.ts";

export type RecipeDto = {
    name: string;
    category: string;
    image: string;
    ingredients: Ingredient[];
    description: string;
    speed: string;
    notes: string;
    opinionOfTheDish: string;
    linkToSource: string;
    favorite: boolean;
}

File: backend/src/main/java/org/example/backend/dto/RecipeDto.java
package org.example.backend.dto;

import lombok.With;
import org.example.backend.model.Ingredient;
import org.example.backend.model.PreparationSpeed;

import java.util.List;

@With
public record RecipeDto(
        String name,
        String category,
        String image,
        List<Ingredient> ingredients,
        String description,
        PreparationSpeed speed,
        String notes,
        String opinionOfTheDish,
        String linkToSource,
        boolean favorite
) {}
